src/pages/index.js

```diff
@@ -1,155 +1,306 @@
+  imagePopup.open(name, link);
+}
 
-// Initialize PopupWithImage, PopupForm instances
 /* -------------------------------------------------------------------------- */
 /*                                Card Section                                */
 /* -------------------------------------------------------------------------- */
-function renderCard(data) {
+function handleConfirmationOpen(cardId, cardElement) {
+  deletePopup.open(cardId, cardElement);
+}
+
+const existingCardIds = new Set();
+
+function createCard(data) {
+  if (existingCardIds.has(data._id)) {
+    return null; // Card already exists, do not create a duplicate
+  }
+
   const card = new Card(
     data,
-    (imageData) => {
-      imagePopup.open(imageData);
-    },
-    "#card-template"
+    handleImageClick,
+    "#card-template",
+    api,
+    handleDelete
   );
-  return card.getView();
+
+  const cardElement = card.getView();
+
+  cardElement.dataset.id = data._id;
+
+  existingCardIds.add(data._id);
+
+  return cardElement;
 }
 
-// Initialize card section
+function handleDelete(cardId, cardElement) {
+  handleConfirmationOpen(cardId, cardElement);
+}
+
+const deletePopup = new PopupWithConfirm(
+  "#verify__card__delete",
+  (cardId, cardElement) => {
+    api
+      .deleteCard(cardId)
+      .then(() => {
+        cardElement.remove();
+        deletePopup.close();
+        existingCardIds.delete(cardId); // Remove from Set when card is deleted
+      })
+      .catch((error) => {
+        console.error("Error deleting card:", error);
+      });
+  }
+);
+
 const cardSection = new Section(
   {
-    items: initialCards,
-    renderer: (data) => {
-      // Create a new card
-      const cardElement = renderCard(data);
-
-      // Display each card
-      cardSection.addItem(cardElement);
+    items: [],
+    renderer: (item) => {
+      const cardElement = createCard(item);
+      if (cardElement) {
+        cardSection.addItem(cardElement);
+      }
     },
   },
-  formSettings.cardList
+  ".cards__list"
 );
 
-// Enable validation for all forms
-
-/* -------------------------------------------------------------------------- */
-/*                                  Add Crad  Form                            */
-/* -------------------------------------------------------------------------- */
-
-// Create the add form instance
-const addCardPopup = new PopupWithForm(formSettings.modalAddCard, (data) => {
-  // Create a new card
-  const newCard = renderCard(data);
-  // Close the add form
-  addCardPopup.close();
-  // Add the new card to the section
-  cardSection.addItem(newCard);
+const addCardPopup = new PopupWithForm("#modalAddCard", (data) => {
+  api.renderCard(data.name, data.link).then((newCardData) => {
+    const cardElement = createCard(newCardData);
+    if (cardElement) {
+      cardSection.addItem(cardElement);
+    }
+    addCardPopup.close();
+  });
 });
+//     .catch((error) => console.error("Error creating new card:", error));
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere**